### PR TITLE
BigBed/BigWig reading and writing

### PR DESCRIPTION
**Pull request by dcjones**
The most elaborate parser yet. This one is for BigBed, which is an indexed, compressed version of Bed. It's indexed in such a way that you can find intervals overlapping a region without reading or parsing the whole file.

```julia
bb = read("motifs.bb", BigBed) # this only reads the file's headers

for interval in bb[Interval("chr9", 141110889, 141150059)]
    # `interval` is of type `BEDInterval`
    @show interval
end
```

This still need's tests, so there's more work to do. I'm focusing on BigBed (and BigWig) right now because my plan is make a package for using the [biodalliance](http://www.biodalliance.org/) browser locally (e.g. within IJulia notebooks). This will let you manipulate and visualize genomics data all within Julia.

### CODE REVIEW

**Comment by dcjones** [comment]
This is ready to go now. Testing is a little light right now, given the complexity of the format, but it's at least checking that BED to BigBed and BigBed to BED conversions are lossless and that intersection queries are correct.

By the way, after some searching I'm pretty sure this only other code to generate BigBed/Wig besides Jim Kent's, so that's pretty cool!

File: src/seq/nucleotide.jl [hunk 1]
```diff
@@ -1518,6 +1545,7 @@ Count ocurrences of short (<= 32) k-mers in a sequence.
 # Arguments:
   * 'seq`: A NucleotideSequence
   * `step`: K-mers counted are separated by this many nucleotides (deafult: 1)
+<<<<<<< HEAD
```
Comment (bicycle1885) [inline src/seq/nucleotide.jl]: A conflict happened here :smile: ?

**Comment by bicycle1885** [comment]
Will `Ragel.Buffer` be superseded by `BufferedReader`?

**Comment by bicycle1885** [comment]
To: @dcjones

**Comment by dcjones** [comment]
I'm actually working on a separate package ([BufferedStreams.jl](https://github.com/dcjones/BufferedStreams.jl)), that I want to use to replace both. I need to do a little more experimentation with that, then I'm going to transfer it to BioJulia and refactor this code.

**Comment by bicycle1885** [comment]
I got it. Thank you :+1:

**Pull request merged**